Content/Items/UndergroundTemple/TempleLensSystem.cs

```diff
@@ -0,0 +1,135 @@
+		public static RenderTarget2D LensTarget;
+
+		public float Priority => 1.1f;
+
+		public void Load()
+		{
+			if (Main.dedServ)
+				return;
+
+			ResizeTarget();
+
+			On.Terraria.Main.DrawNPCs += DrawLens;
+			On.Terraria.Main.CheckMonoliths += Main_CheckMonoliths;
+		}
+
+		public void Unload()
+		{
+			if (Main.dedServ)
+				return;
+
+			On.Terraria.Main.DrawNPCs -= DrawLens;
+			On.Terraria.Main.CheckMonoliths -= Main_CheckMonoliths;
+		}
+
+		private void Main_CheckMonoliths(On.Terraria.Main.orig_CheckMonoliths orig)
+		{
+			GraphicsDevice gD = Main.graphics.GraphicsDevice;
+			SpriteBatch spriteBatch = Main.spriteBatch;
+
+			if (Main.gameMenu || Main.dedServ || spriteBatch is null || NPCTarget is null || gD is null || (!Main.npc.Any(n => n.active && n.HasBuff(ModContent.BuffType<Buffs.Illuminant>()))) && !Main.npc.Any(n => n.active && n.HasBuff(ModContent.BuffType<Exposed>())))
+				return;
+
+			#region npctarget
+			RenderTargetBinding[] bindings = gD.GetRenderTargets();
+			gD.SetRenderTarget(NPCTarget);
+			gD.Clear(Color.Transparent);
+
+			Main.spriteBatch.Begin(default, default, default, default, default, null, Main.GameViewMatrix.ZoomMatrix);
+
+			for (int i = 0; i < Main.npc.Length; i++)
+			{
+				NPC NPC = Main.npc[i];
+
+				if (NPC.active)
+				{
+					if (NPC.ModNPC != null)
+					{
+						if (NPC.ModNPC != null && NPC.ModNPC is ModNPC ModNPC)
+						{
+							if (ModNPC.PreDraw(spriteBatch, Main.screenPosition, NPC.GetAlpha(Color.White)))
+								Main.instance.DrawNPC(i, false);
+
+							ModNPC.PostDraw(spriteBatch, Main.screenPosition, NPC.GetAlpha(Color.White));
+						}
+					}
+					else
+					{
+						Main.instance.DrawNPC(i, false);
+					}
+				}
+			}
+
+			spriteBatch.End();
+
+			gD.SetRenderTarget(LensTarget);
+			gD.Clear(Color.Transparent);
+
+			Main.spriteBatch.Begin(default, BlendState.AlphaBlend, default, default, default, null, Main.GameViewMatrix.ZoomMatrix);
+
+			Texture2D bloom = ModContent.Request<Texture2D>(AssetDirectory.Keys + "GlowAlpha").Value;
+			Color gold = Color.Orange;
+			gold.A = 0;
+			for (int i = 0; i < Main.npc.Length; i++)
+			{
+				NPC NPC = Main.npc[i];
+
+				if (NPC.active && NPC.HasBuff(ModContent.BuffType<Buffs.Illuminant>()))
+				{
+					spriteBatch.Draw(bloom, NPC.Center - Main.screenPosition, null, gold, 0, bloom.Size() / 2, 5, SpriteEffects.None, 0f);
+				}
+
+				if (NPC.active && NPC.HasBuff(ModContent.BuffType<Exposed>()))
+				{
+					spriteBatch.Draw(bloom, NPC.Center - Main.screenPosition, null, gold, 0, bloom.Size() / 2, 2, SpriteEffects.None, 0f);
+				}
+			}
+			spriteBatch.End();
+			gD.SetRenderTargets(bindings);
+			#endregion
+		}
+
+		private void DrawLens(On.Terraria.Main.orig_DrawNPCs orig, Main self, bool behindTiles)
+		{
+			orig(self, behindTiles);
+			if (!behindTiles && (Main.npc.Any(n => n.active && n.HasBuff(ModContent.BuffType<Buffs.Illuminant>())) || Main.npc.Any(n => n.active && n.HasBuff(ModContent.BuffType<Exposed>()))))
```

Comment:
Should we consider making a static property for this so we dont have to copy/paste this complex condition around?